src/main/java/oracle/kubernetes/operator/DomainWatcher.java

```diff
@@ -73,14 +73,9 @@ public void run() {
        */
       @Override
       public Watch<Domain> initiateWatch(Object context, String resourceVersion) throws ApiException {
-        return Watch.createWatch(client.getApiClient(),
-            client.callBuilder().with($ -> {
-              $.resourceVersion = resourceVersion;
-              $.timeoutSeconds = 30;
-              $.watch = true;
-            }).listDomainCall(ns),
-            new TypeToken<Watch.Response<Domain>>() {
-            }.getType());
+        return new WatchBuilder(client)
+                  .withResourceVersion(resourceVersion)
+                .createDomainsInNamespaceWatch(ns);
```

Comment:
You call this createDomainsInNamespaceWatch, but the Pod, Ingress, and Service versions are just createXXXWatch().  Why is this?  I'm guessing it's because you have an *all* namespaces watch variant, correct?  That variant would never be interesting to the runtime so we should probably refactor the test code to not use it either.

Reply:
I largely modeled the createXXX methods after the ones in CallBuilder. I am certainly open to changing them.

Reply:
Yes, I see that...  My only comment is about watching domains -- you have two variants: one for a single namespace and the other for all namespaces.  I'm guessing you did this because the test watches for domains in all namespaces, but I'd prefer to change the test since there is no runtime use case for watching domains in *all* namespaces.